### PR TITLE
Add cohere processor for text generation.

### PR DESCRIPTION
**Pull request by rubio-He**
#115  Will send another PR to add CohereGenerateProcessorConfig.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -141,6 +141,8 @@ dependencies {
     implementation 'org.yaml:snakeyaml:1.33'
     // H2 in-memory database
     testImplementation 'com.h2database:h2:1.4.200'
+    // Guice dependency injection
+    testImplementation group: 'com.google.inject.extensions', name: 'guice-testlib', version: '5.1.0'
```
Comment (li2109) [inline build.gradle:145]: using the compact way?
like: testImplementation  'com.google.inject.extensions:guice-testlib:5.1.0'?

File: src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  @Inject private CohereGenerateProcessor cohereGenerateProcessor;
+
+  @BeforeEach
+  public void setUp() {
+    Guice.createInjector(BoundFieldModule.of(this)).injectMembers(this);
+  }
+
+  @Test
+  void run() {
+    // Prepare
```
Comment (li2109) [inline src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java:38]: // Arrange.

File: src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    Generation generation = new Generation();
+    generation.setText("This ice cream is sweet and delicious");
+    CohereGenerateResponse generateResponse = new CohereGenerateResponse();
+    generateResponse.setGenerations(ImmutableList.of(generation));
+    when(CohereApiService.generate(any())).thenReturn(generateResponse);
+
+    SingleText input = SingleText.of("ice cream");
+
+    // Act
+    SingleText actualOutput = cohereGenerateProcessor.run(input);
```
Comment (li2109) [inline src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java:48]: nit: period at the end => // Act.

File: src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  void runAsync() throws Exception {
+    // Prepare
+    Generation generation = new Generation();
+    generation.setText("This ice cream is sweet and delicious");
+    CohereGenerateResponse generateResponse = new CohereGenerateResponse();
+    generateResponse.setGenerations(ImmutableList.of(generation));
+    when(CohereApiService.generateAsync(any()))
+        .thenReturn(Futures.immediateFuture(generateResponse));
+
+    SingleText input = SingleText.of("ice cream");
```
Comment (li2109) [inline src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java:64]: cohereGenerateProcessor.runAsync(SingleText.of("ice cream")); ?

File: src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import com.google.inject.Provides;
+
+import java.time.Duration;
+
+import static ai.knowly.langtorch.llm.cohere.CohereApiService.buildApi;
+
+public final class CohereProcessorModule extends AbstractModule {
+  private static final Duration DEFAULT_TIMEOUT = Duration.ofSeconds(10);
+
+  private static FluentLogger logger = FluentLogger.forEnclosingClass();
```
Comment (li2109) [inline src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java:17]: private static final

File: src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import ai.knowly.langtorch.utils.api.key.CohereKeyUtil;
+import com.google.common.flogger.FluentLogger;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+
+import java.time.Duration;
+
+import static ai.knowly.langtorch.llm.cohere.CohereApiService.buildApi;
+
+public final class CohereProcessorModule extends AbstractModule {
```
Comment (li2109) [inline src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java:14]: I was originally using the Guice module too but I was also wondering later on: does this mean we have to enforce our users to use Guice as the dependency injection framework?
but it's okay if it's only for testing.

**Review by li2109 — CHANGES_REQUESTED** [review]

File: src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java (hunk 6 above)
Comment (li2109) [reply src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java:14]: yea, we should go with Guice.
https://github.com/Knowly-ai/langtorch/issues/119

File: build.gradle (hunk 1 above)
Comment (rubio-He) [reply build.gradle:145]: Sounds good!

File: src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java (hunk 2 above)
Comment (rubio-He) [reply src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java:38]: Done!

File: src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java (hunk 4 above)
Comment (rubio-He) [reply src/test/java/ai/knowly/langtorch/processor/cohere/generate/CohereGenerateProcessorTest.java:64]: Done!

File: src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java (hunk 5 above)
Comment (rubio-He) [reply src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java:17]: Done!

**Comment by li2109** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=Knowly-ai_langtorch&pullRequest=118)
> 
> [Bug](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=Knowly-ai_langtorch&pullRequest=118&metric=new_coverage&view=list) [0.0% Coverage](https://sonarcloud.io/component_measures?id=Knowly-ai_langtorch&pullRequest=118&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=Knowly-ai_langtorch&pullRequest=118&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=Knowly-ai_langtorch&pullRequest=118&metric=new_duplicated_lines_density&view=list)

Please also resolve this warning. Should be fairly easy:
public UpsertResponse addDocuments(List<DomainDocument> documents) {
return addVectors(
        documents.stream()
            .map(this::createVector)
            .collect(Collectors.toCollection(ArrayList::new)));
  }

**Comment by rubio-He** [comment]
> > Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=Knowly-ai_langtorch&pullRequest=118)
> > [Bug](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=Knowly-ai_langtorch&pullRequest=118&resolved=false&types=CODE_SMELL)
> > [](https://sonarcloud.io/component_measures?id=Knowly-ai_langtorch&pullRequest=118&metric=new_coverage&view=list) [0.0% Coverage](https://sonarcloud.io/component_measures?id=Knowly-ai_langtorch&pullRequest=118&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=Knowly-ai_langtorch&pullRequest=118&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=Knowly-ai_langtorch&pullRequest=118&metric=new_duplicated_lines_density&view=list)
> 
> Please also resolve this warning. Should be fairly easy: public UpsertResponse addDocuments(List documents) { return addVectors( documents.stream() .map(this::createVector) .collect(Collectors.toCollection(ArrayList::new))); }

Noticed that you resolved the code smell.

**Pull request merged**